File: src/main/java/br/com/siscake/service/UsuarioService.java
package br.com.siscake.service;

import java.util.List;

import br.com.siscake.modelo.Usuario;

public interface UsuarioService {

	/**
	 * Método responsável gravar o usuário na base de dados
	 * @param usuario
	 */
	public void saveUsuario(Usuario usuario);
	
	
	/**
	 * Método responsável alterar o usuário na base de dados
	 * @param usuario
	 */
	public void updateUsuario(Usuario usuario);
	
	/**
	 * Método resposável por retornar uma lista de usuários de acordo com os parâmetros.
	 * 
	 * @param usuario
	 * @return Lista de usuário - List
	 */
	public List<Usuario> findUsuarios(Usuario usuario);
	
	/**
	 * Método responsável por retornar um usuario da base dados pelo código.
	 * 
	 * @param idUsuario - Long
	 * @return retorna um Usuário - 
	 */
	public Usuario findUsuarioById(Long idUsuario);
	
	/**
	 * Método responsável por excluir da base de dados o usuário, de acordo com o id passado.
	 * 
	 * @param idUsuario
	 */
	public void deleteUsuario(Long idUsuario);

}

File: src/main/webapp/app/manterUsuario/services/usuarioService.js
app.service('UsuarioService', function(Restangular){

	var urlBase = "manterUsuario/";
	var usuario = {};
	
	/**
	 * Reponsável por salvar ou alterar
	 * 
	 * @param usuario
	 * 
	 */
	this.salvarAlterar = function(usuario){
		Restangular.all(urlBase+"saveUsuario").post(usuario).then(function() {
            console.log("Usuario salvo");
        });
	}
	
	
	/**
	 * Responsável por retorar um usuário.
	 * 
	 * @param id - código de consulta do objeto no backend.
	 * @param callback - funcao que é executada no controler, quando retornar os dados  
	 */
	this.findById = function(id, callback){
		
		//Pega o id do parametro passado na url, vai no back-end e retorna o dado
		Restangular.one(urlBase+"findUsuarioById", id).get()
		.then(callback);
	}
	
	/**
	 * Resposável por retornar um usuario de acordo com o id.
	 * 
	 *  @param id - código de consulta do objeto no backend.
	 *  
	 *  @return retorna uma promessa.
	 * 
	 */
	this.findByIdThen = function(id){
		
		//Pega o id do parametro passado na url, vai no back-end e retorna o dado
		return Restangular.one(urlBase+"findUsuarioById", id).get();
	}
	
	this.usuarioVazio = function(){
		var usuarioVazio = {
		        id: null,
		        nome: "",
		        cpf: "",
		        dsEmail:"",
		        senha:""
			}
		
		return usuarioVazio;
	}
	
});